src/main/java/carpet/CarpetSettings.java

```diff
@@ -344,8 +344,32 @@ public String description() {
     )
     public static boolean antiCheatDisabled = false;
 
-    @Rule(desc = "Pistons, droppers and dispensers react if block above them is powered", category = CREATIVE)
-    public static boolean quasiConnectivity = true;
+    private static class QuasiConnectivityValidator extends Validator<Integer> {
+
+        @Override
+        public Integer validate(CommandSourceStack source, CarpetRule<Integer> changingRule, Integer newValue, String userInput) {
+            int minRange = 1;
```

Comment:
Given the rule now replaces qc false the minimum should be zero.
```suggestion
            int minRange = 0;
```